lib/msf/core/post/windows/priv.rb

```diff
@@ -356,14 +357,15 @@ def decrypt_lsa_data(policy_secret, lsa_key)
     end
 
     aes = OpenSSL::Cipher.new("aes-256-cbc")
+    aes.decrypt
     aes.key = sha256x.digest
 
     vprint_status("digest #{sha256x.digest.unpack("H*")[0]}")
 
     decrypted_data = ''
 
     (60...policy_secret.length).step(16) do |i|
-      aes.decrypt
+      aes.reset
```

Comment:
This file was the only use of openssl that was being tested and impacted by the update

Reply:
I thought this was weird at the time having to reset this as part of the loop, as it goes against how `aes-256-cbc` is meant to function, i.e. the cipher blocks are chained together.

I'm thinking this is meant to be `aes-256-ecb` maybe? It passes the tests still:

```diff
-    aes = OpenSSL::Cipher.new("aes-256-cbc") 
+    aes = OpenSSL::Cipher.new("aes-256-ecb")
     aes.decrypt
+    aes.padding = 0
     aes.key = sha256x.digest

     vprint_status("digest #{sha256x.digest.unpack("H*")[0]}")

     decrypted_data = ''

     (60...policy_secret.length).step(16) do |i|
-      aes.reset
-      aes.padding = 0
       decrypted_data << aes.update(policy_secret[i,16])
     end

    return decrypted_data
```

Reply:
I can't seem to find any official Microsoft documentation to explain this. The way Metasploit does it is very similar to how impacket handles it, see [secretsdump.py:1081](https://github.com/SecureAuthCorp/impacket/blob/e1b9e247de5b4514ea3f5b0229e8187d9e8142d4/impacket/examples/secretsdump.py#L1081:9). Another implementation that I found is [RemoteHashRetrieval.ps1:943](https://github.com/HarmJ0y/DAMP/blob/master/RemoteHashRetrieval.ps1#L943) which does not appear to reset the cipher.

Reply:
I'm happy to leave this as-is just in case there's weird edge cases that aren't immediately obvious 👍

Reply:
Yeah I spent a few hours looking through lsasrv.dll trying to find the exact encryption algorithm and was unsuccessful. Working in reverse, assuming that the CNG API is used, I only found references to CFB and CBC mode being used with AES encryption. I'm not entirely sure this is what's used though, I'd need to do more analysis.

![image](https://user-images.githubusercontent.com/58950994/110041303-ebdd5880-7d11-11eb-9def-cefe8a58a01d.png)


Leaving it in its current state is probably our best bet.

Reply:
Thanks for taking a deeper look! 🎉